src/main/java/uk/co/ramp/covid/simulation/population/Adult.java

```diff
@@ -46,37 +45,25 @@ public void reportDeath(DailyStats s) {
     }
 
     @Override
-    public void allocateCommunalPlace(Population p) {
+    public void allocateCommunalPlace(Places p) {
         switch(profession) {
             case TEACHER: {
-                CommunalPlace property = p.getRandomPlace();
-                while (!(property instanceof School)) property = p.getRandomPlace();
-                setPrimaryPlace(property);
+                setPrimaryPlace(p.getRandomSchool());
             } break;
             case SHOP: {
-                CommunalPlace property = p.getRandomPlace();
-                while (!(property instanceof Shop)) property = p.getRandomPlace();
-                setPrimaryPlace(property);
+                setPrimaryPlace(p.getRandomShop());
             } break;
             case CONSTRUCTION: {
-                CommunalPlace property = p.getRandomPlace();
-                while (!(property instanceof ConstructionSite)) property = p.getRandomPlace();
-                setPrimaryPlace(property);
+                setPrimaryPlace(p.getRandomConstructionSite());
             } break;
             case OFFICE: {
-                CommunalPlace property = p.getRandomPlace();
-                while (!(property instanceof Office)) property = p.getRandomPlace();
-                setPrimaryPlace(property);
+                setPrimaryPlace(p.getRandomOffice());
             } break;
             case HOSPITAL: {
-                CommunalPlace property = p.getRandomPlace();
-                while (!(property instanceof Hospital)) property = p.getRandomPlace();
-                setPrimaryPlace(property);
+                setPrimaryPlace(p.getRandomHospital());
             } break;
             case RESTAURANT: {
-                CommunalPlace property = p.getRandomPlace();
-                while (!(property instanceof Restaurant)) property = p.getRandomPlace();
-                setPrimaryPlace(property);
+                setPrimaryPlace(p.getRandomRestaurant());
             } break;
```

Comment:
Do you need a case NONE for unemployed people?

Reply:
I don't think so since we do a null check wherever the primary place is used (since for example Pensioners never have a primary place). There's an argument that we should make this an Optional to highlight this (and make sure at the type we check for a value; I'll add this as a possible further refactoring since this change is very large already.)